libnavigation-core/src/test/java/com/mapbox/navigation/core/routealternatives/AlternativeRouteProgressDataProviderTest.kt

```diff
@@ -0,0 +1,289 @@
+            listOf(
+                listOf(List(10) { mockk() }, List(7) { mockk() }, List(15) { mockk() }),
+                listOf(List(23) { mockk() }),
+                listOf(List(32) { mockk() }, List(19) { mockk() }),
+                listOf(List(14) { mockk() }, List(23) { mockk() }), // 135 points for legs #0-#3
+                listOf(List(8) { mockk() }),
+            )
+        )
+        val expected = RouteProgressData(4, 200, 65)
+
+        val actual = AlternativeRouteProgressDataProvider.getRouteProgressData(
+            primaryRouteProgressData,
+            alternativeMetadata
+        )
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `before fork, alternative starts after the primary route on the first leg`() =
+        coroutineRule.runBlockingTest {
+            val primaryRouteProgressData = RouteProgressData(4, 200, 35)
+            val alternativeMetadata = alternativeRouteMetadata(
+                primaryForkLegIndex = 5,
+                primaryForkRouteGeometryIndex = 250,
+                primaryForkLegGeometryIndex = 85,
+                alternativeForkLegIndex = 5,
+                alternativeForkRouteGeometryIndex = 220,
+                alternativeForkLegGeometryIndex = 86,
+                listOf(
+                    listOf(List(10) { mockk() }, List(7) { mockk() }, List(15) { mockk() }),
+                    listOf(List(23) { mockk() }),
+                    listOf(List(32) { mockk() }, List(19) { mockk() }),
+                    listOf(List(14) { mockk() }, List(23) { mockk() }), // 135 points for legs #0-#3
+                    listOf(List(8) { mockk() }),
+                )
+            )
+            val expected = RouteProgressData(4, 170, 35)
+
+            val actual = AlternativeRouteProgressDataProvider.getRouteProgressData(
+                primaryRouteProgressData,
+                alternativeMetadata
+            )
+
+            assertEquals(expected, actual)
+        }
+
+    @Test
+    fun `before fork, alternative starts after the primary route on the second leg`() =
+        coroutineRule.runBlockingTest {
+            val primaryRouteProgressData = RouteProgressData(4, 200, 35)
+            val alternativeMetadata = alternativeRouteMetadata(
+                primaryForkLegIndex = 5,
+                primaryForkRouteGeometryIndex = 250,
+                primaryForkLegGeometryIndex = 85,
+                alternativeForkLegIndex = 3,
+                alternativeForkRouteGeometryIndex = 220,
+                alternativeForkLegGeometryIndex = 81,
+                listOf(
+                    listOf(List(10) { mockk() }, List(7) { mockk() }, List(15) { mockk() }),
+                    listOf(List(23) { mockk() }), // 135 points for legs #0-#1
+                    listOf(List(32) { mockk() }, List(19) { mockk() }),
+                    listOf(List(14) { mockk() }, List(23) { mockk() }),
+                    listOf(List(8) { mockk() }),
```

Comment:
```suggestion
                    listOf(List(10) { mockk() }, List(7) { mockk() }, List(15) { mockk() }),
                    listOf(List(23) { mockk() }), // 52 points for legs #0-#1
                    listOf(List(32) { mockk() }, List(19) { mockk() }),
```
The test route has only 3 legs. Can be addressed separately.